automation_tools/scripts/ga-migration/main.py

```diff
@@ -0,0 +1,238 @@
+    # import wdb; wdb.set_trace()
+    if not os.path.isfile(filepath):
+        logging.info("No %s found" % filepath)
+    else:
+        logging.info("Found %s" % filepath)
+        with open(filepath, "r") as f:
+            lines = f.readlines()
+        with open(filepath, "w") as f:
+            for line in lines:
+                if term not in line:
+                    f.write(line)
+                else:
+                    logging.info(f"TASK: Line deleted")
+
+
+def file_contains(term, filepath):
+    """Check whether file contains given term."""
+    with open(filepath) as f:
+        return term in f.read()
+
+
+def append_to_file(text, filepath):
+    """Append text to file."""
+    with open(filepath, "a") as f:
+        f.write(text)
+
+
+def add_line(term, filepath):
+    """ Add a line to a file """
+    logging.info("TASK: Adding line '%s' to %s" % (term, filepath))
+    # If the file exists
+    if os.path.isfile(filepath):
+        # And the line is not already there
+        if not file_contains(term, filepath):
+            append_to_file(term, filepath)
+        else:
+            logging.info("SKIPPED TASK. Line already there. ")
+    else:
+        logging.info("SKIPPED TASK. No %s found" % filepath)
+
+
+def replace_simple(text, replacing, filepath):
+    """
+    Replaces every match of a string with another in the specified file
+    """
+    logging.info(
+        "TASK: Simple replacing %s with %s in %s" % (text, replacing, filepath)
+    )
+    if os.path.isfile(filepath):
+        logging.info("Found %s" % filepath)
+        # TODO: expose number of matches
+        with fileinput.FileInput(filepath, inplace=True, backup=".bak") as file:
+            for line in file:
+                print(line.replace(text, replacing), end="")
+    else:
+        logging.info("No %s found" % filepath)
+
+
+def replace_regex(regex, output, filepath):
+    """
+    Replaces every match of a string with another in the specified file
+    """
+    logging.info(
+        "TASK: RegEx replacing %s with %s in %s" % (regex, output, filepath)
+    )
+    if os.path.isfile(filepath):
+        logging.info("Found %s" % filepath)
+        # TODO: expose number of matches
+        with fileinput.FileInput(filepath, inplace=True, backup=".bak") as file:
+            for line in file:
+                print(re.sub(regex, output, line), end="")
+    else:
+        logging.info("SKIPPED TASK. No %s found" % filepath)
+
+
+def download_file(url, destination):
+    # Get path
+    dirname = os.path.dirname(os.path.realpath(destination))
+    if not os.path.exists(dirname):
+        os.makedirs(dirname)
+    print(dirname)
+    r = requests.get(url, allow_redirects=True)
+    open(destination, "wb").write(r.content)
+
+
+def replace_list(filepath, regex, to_remove, to_add, var_name):
+    with open(filepath, "r") as f:
+        contents = f.read()
+
+    m = re.search(regex, contents)
+
+    matched_list_str = m.group(1)
```

Comment:
In your example it works, but I've tried with a real repo and I get:

```console
automation_tools/scripts/ga-migration/main.py:125: FutureWarning: Possible nested set at position 18
  m = re.search(regex, contents)
Traceback (most recent call last):
  File "automation_tools/scripts/ga-migration/main.py", line 238, in <module>
    pipeline()
  File "/Users/marco/.virtualenvs/automation-tools/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/marco/.virtualenvs/automation-tools/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/marco/.virtualenvs/automation-tools/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/marco/.virtualenvs/automation-tools/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "automation_tools/scripts/ga-migration/main.py", line 228, in pipeline
    replace_list(
  File "automation_tools/scripts/ga-migration/main.py", line 127, in replace_list
    matched_list_str = m.group(1)
AttributeError: 'NoneType' object has no attribute 'group'
```

Can you reproduce it? It seems that the regex is not catching anything.

Reply:
I've got this error too in `invenio-mail`.

Reply:
Perhaps we don't need the `stub` anymore and we just test it with real repos?

Reply:
Looks like it doesn't fail anymore with `invenio-mail`